website/docs/advanced-concepts/advanced-integrations.md

```diff
@@ -0,0 +1,133 @@
+---
+sidebar_position: 0
+---
+
+# Integration - Advanced
+
+This section goes into detail on how [Integrations](../concepts/integration/) work. If you are just looking to use Functionless without extending it, this section can be skipped. If you have not read [Integrations](../concepts/integration/), do that first.
+
+## Terms
+
+- Resource - Infrastructure and Business Logic which use an integration, generally modelled as some infrastructure like a [`Function`](../concepts/function/) or [`EventBus`](../concepts/event-bridge/).
+- [Integration](../concepts/integration/) - An object (may be a Resource) or function which can be used within a Resource.
+- [Integration Type](./integration-types.md) - A pattern that an integration implements to be used within one or more Resources.
+
+```mermaid
+graph LR;
+  IntegrationType(Integration Type)
+  Resource--uses-->Integration;
+  Resource--supports-->IntegrationType;
+  Integration--implements-->IntegrationType;
+```
+
+A view of what a user sees of Integrations and Resources:
+
+```ts
+// event bus is a Resource
+const bus = new EventBus<Event<{ name: string }>>(this, "bus");
+
+// StepFunction is a Resource
+const sfn = new StepFunction(this, sfn, (payload: { name: string }) => {
+  // bus.putEvents is an Integration that supports the StepFunction ASL Integration Type
+  bus.putEvents({
+    source: "specialSource",
+    "detail-type": "UserNameEvent",
+    detail: payload,
+  });
+});
+```
+
+Partial view of `EventBus` and `putEvents` which supports the StepFunction Integration Type:
+
+```ts
+class EventBus {
+
+  public readonly putEvents: IntegrationCall<
+    "EventBus.putEvents",
+    IEventBus<E>["putEvents"]
+  >;
```

Comment:
This is quite a complex type

Reply:
Yeah... it was hard to find simple examples to show for this documentation.

Options:
1. Remove this, it isn't necessary 
2. Don't show the real types
3. leave it
4. Expand the types, which are more complex, but more explicit.

Going with option 1

Reply:
You can use inference on the class member. Just omit the type from the declaration